src/safeposix/filesystem.rs

```diff
@@ -108,30 +105,96 @@ impl FilesystemMetadata {
 
         retval
     }
-}
-
-pub fn load_fs() {
 
-    // Create initial cage, probably will move this
-    let utilcage = Cage{cageid: 0,
-        cwd: interface::RustLock::new(interface::RustRfc::new(interface::RustPathBuf::from("/"))),
-        parent: 0, 
-        filedescriptortable: interface::RustLock::new(interface::RustHashMap::new()),
-        getgid: interface::RustAtomicI32::new(-1), 
-        getuid: interface::RustAtomicI32::new(-1), 
-        getegid: interface::RustAtomicI32::new(-1), 
-        geteuid: interface::RustAtomicI32::new(-1)
-    };
+    // Read file, and deserialize CBOR to FS METADATA
+    pub fn init_fs_metadata() -> FilesystemMetadata {
+        // Read CBOR from file
+        if interface::pathexists(METADATAFILENAME.to_string()) {
+            let metadata_fileobj = interface::openfile(METADATAFILENAME.to_string(), false).unwrap();
+            let metadatabytes = metadata_fileobj.readfile_to_new_bytes().unwrap();
+            metadata_fileobj.close().unwrap();
+    
+            // Restore metadata
+            interface::serde_deserialize_from_bytes(&metadatabytes).unwrap()
+        } else {
+            FilesystemMetadata::blank_fs_init()
+        }
+    }
+}
 
-    let mut mutmetadata = FS_METADATA.write().unwrap();
+pub fn format_fs() {
```

Comment:
I see how this works. Nice solution... unfortunate that we had to do everything manually, though.